docs/building_the_project_with_dpcpp.rst

```diff
@@ -0,0 +1,491 @@
+
+* On Linux*
+
+.. code-block:: bash
+
+   # Inside <path to onemkl>
+   mkdir build && cd build
+   cmake .. -DCMAKE_CXX_COMPILER=$CXX_COMPILER \ # Should be icpx or clang++
+            -DCMAKE_C_COMPILER=$C_COMPILER \ # Should be icx or clang
+            -DENABLE_PORTFFT_BACKEND=ON \
+            -DENABLE_MKLCPU_BACKEND=OFF  \
+            -DENABLE_MKLGPU_BACKEND=OFF  \
+            [-DPORTFFT_REGISTERS_PER_WI=128] \ # Example portFFT tuning parameter
+            [-DREF_BLAS_ROOT=<reference_blas_install_prefix>] \ # required only for testing
+            [-DPORTFFT_DIR=<path to portFFT install directory>]
+   cmake --build .
+   ctest
+
+
+portFFT will be downloaded automatically if not found.
+
+By default, the portFFT backend is not tuned for any specific device. The tuning flags are
+detailed in the `portFFT <https://github.com/codeplaysoftware/portFFT>`_ repository.
+The tuning parameters can be set at configuration time,
+with the above example showing how to set the tuning parameter
+``PORTFFT_REGISTERS_PER_WI``. Note that some tuning configurations may be incompatible
+with some targets.
+
+The portFFT library is compiled using the same ``-fsycl-targets`` as specified
+by the ``CMAKE_CXX_FLAGS``. If none are found, it will compile for
+``-fsycl-targets=nvptx64-nvidia-cuda,spir64``. To enable HIP targets,
+``HIP_TARGETS`` must be specified. See
+`DPC++ User Manual <https://intel.github.io/llvm-docs/UsersManual.html>`_
+for more information on ``-fsycl-targets``.
+
+.. _build_additional_options_dpcpp:
+
+Additional Build Options
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When building oneMKL the SYCL implementation can be specified by setting the
+``ONEMKL_SYCL_IMPLEMENTATION`` option. Possible values are:
+
+* ``dpc++`` (default) for the
+  `Intel(R) oneAPI DPC++ Compiler <https://software.intel.com/en-us/oneapi/dpc-compiler>`_
+  and for the
+  `oneAPI DPC++ Compiler <https://github.com/intel/llvm>`_ compilers.
+* ``hipsycl`` for the `AdaptiveCpp <https://github.com/illuhad/AdaptiveCpp>`_ SYCL implementation.
+Please see `building with AdaptiveCpp` if using this option.
+
+The following table provides details of CMake options and their default values:
+
+.. list-table::
+   :header-rows: 1
+
+   * - CMake Option
+     - Supported Values
+     - Default Value 
+   * - BUILD_SHARED_LIBS
+     - True, False
+     - True      
+   * - BUILD_DOC
+     - True, False
+     - False     
+
+
+.. note::
+  When building with clang++ for AMD backends, you must additionally set
+  ``SYCL_DEVICE_FILTER`` to ``HIP`` and provide ``-DHIP_TARGETS`` according to
+  the targeted hardware. This backend has only been tested for the ``gfx90a``
+  architecture (MI210) at the time of writing. 
+
+.. note::
+  When building with ``BUILD_FUNCTIONAL_TESTS=yes`` (default option) only single CUDA backend can be built
+  (`#270 <https://github.com/oneapi-src/oneMKL/issues/270>`_).
+
+
+.. _building_and_running_tests_dpcpp:
+
+Building and running tests
```

Comment:
Should this section also be included in the AdaptiveCpp document? Or should we make it a separate file?

Reply:
I've put this in a separate file in [2b9a352](https://github.com/oneapi-src/oneMKL/pull/510/commits/2b9a352c5755b09fc8c395233f1f4393e54fdbe3).